components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/ExceptionCodes.java

```diff
@@ -544,6 +544,9 @@ public enum ExceptionCodes implements ErrorHandler {
      * @param errorDescription The error description.
      */
     ExceptionCodes(long errorCode, String msg, int httpErrorCode, String errorDescription) {
+        if(httpErrorCode>=500){
+                this.stackTrace=true;
+        }
```

Comment:
Shouldn't the fix be to use the other constructor which accepts the `stackTrace` flag for 5xx errors? 🤔

Reply:
Ideally we need the stack traces for Server Errors, which are 5xx errors.
Other method can also be used, but that is causing changes to all the exception code initialization which I think is unnecessary as we can do the same thing in a simple logic as above.